src/game/SpellEffects.cpp

```diff
@@ -1149,6 +1149,13 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
 
                     return;
                 }
+                case 32027:                                 // Expedition Flare
+                {
+                    if (!m_CastItem)                        // 32029 = Expedition Preserver | 32030 = Expedition Scout
```

Comment:
You're right, it's fine to send a nullptr, nothing bad can happen, well, the spell may not function properly, but in this case it propably will since blizz must have run into the same kind of thing;
This was also my suggestion, I just wasn't aware that the item was consumed on use, now that I know it is, then removing the check doesn't seem all that wrong to me anymore.